src/components/Card.js

```diff
@@ -1,47 +1,99 @@
-class Card {
-  constructor({ name, link }, cardTemplateSelector, handleCardClick) {
+export default class Card {
+  constructor(
+    { name, link, _id, isLiked },
+    cardTemplateSelector,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
     this._name = name;
     this._link = link;
+    this._id = _id;
+    this._isLiked = isLiked;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
     this.cardTemplate =
       document.querySelector(cardTemplateSelector).content.firstElementChild;
+    this._modal = document.querySelector("#card-delete-modal");
+    this._handleEscClose = this._handleEscClose.bind(this); // Bind the method here
   }
 
+  // Open the modal
+  open() {
+    this._modal.classList.add("modal_opened");
+    document.addEventListener("keydown", this._handleEscClose); // Listen for ESC key
+  }
+
+  // Close the modal
+  close() {
+    this._modal.classList.remove("modal_opened");
+    document.removeEventListener("keydown", this._handleEscClose); // Stop listening for ESC
+  }
+
+  // Remove the card element
+  removeCard() {
+    this._cardElement.remove();
+    this._cardElement = null;
+  }
+
+  // Get the card element
   getElement() {
     this._cardElement = this.cardTemplate.cloneNode(true);
     const cardImageEl = this._cardElement.querySelector(".card__image");
     const cardTitleEl = this._cardElement.querySelector(".card__title");
+    this._likeButton = this._cardElement.querySelector(".card__like-button"); // Assign it here
+    this._deleteBtn = this._cardElement.querySelector(".card__delete-button");
 
     cardImageEl.src = this._link;
     cardImageEl.alt = this._name;
     cardTitleEl.textContent = this._name;
 
-    this._setEventListeners();
+    // If the card is liked, make the like button active
+    if (this._isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    }
+
+    this._setEventListeners(); // Set event listeners
     return this._cardElement;
   }
 
+  // Set event listeners for the card
   _setEventListeners() {
-    const cardLikeButton =
-      this._cardElement.querySelector(".card__like-button");
     const trashButton = this._cardElement.querySelector(".trash-button");
 
     trashButton.addEventListener("click", () => {
-      this._cardElement.remove();
+      this._handleDeleteClick(this); // Pass the card instance to the delete handler
     });
 
-    cardLikeButton.addEventListener("click", () => {
-      cardLikeButton.classList.toggle("card__like-button_active");
+    const modalCloseButton = this._modal.querySelector(
+      "#card-delete-close-btn"
+    ); // Access the modal close button
+
+    this._likeButton.addEventListener("click", () => {
+      this._likeButton.classList.toggle("card__like-button_active");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.